utils/validators.test.ts

```diff
@@ -0,0 +1,104 @@
+import * as v from './validators'
+
+// region: test `validateAddresses`
+
+test('Accept example addresses', () => {
+  expect(v.validateAddresses(
+    ['ecash:qpz274aaj98xxnnkus8hzv367za28j900c7tv5v8pc',
+     'bitcoincash:qz0dqjf6w6dp0lcs8cc68s720q9dv5zv8cs8fc0lt4',]
+  )).toBe(true)
+});
+
+
+test('Accept addresses for testnets', () => {
+  expect(v.validateAddresses(
+    ['bchreg:qpz274aaj98xxnnkus8hzv367za28j900c7tv5v8pc',
+     'bchtest:qz0dqjf6w6dp0lcs8cc68s720q9dv5zv8cs8fc0lt4',]
+  )).toBe(true)
+});
+
+test('Accept example addresses uppercase', () => {
+  expect(v.validateAddresses(
+    ['ecash:QPZ274AAJ98XXNNKUS8HZV367ZA28J900C7TV5V8PC',
+     'bitcoincash:QZ0DQJF6W6DP0LCS8CC68S720Q9DV5ZV8CS8FC0LT4',]
+  )).toBe(true)
+});
+
+test('Accept example address one uppercase other lowercase', () => {
+  expect(v.validateAddresses(
+    ['ecash:QPZ274AAJ98XXNNKUS8HZV367ZA28J900C7TV5V8PC',
+     'bitcoincash:qz0dqjf6w6dp0lcs8cc68s720q9dv5zv8cs8fc0lt4',]
+  )).toBe(true)
+});
+test('Reject chain uppercase', () => {
```

Comment:
Sorry man, I forgot to mention this from the spec. We should squash the chain to lowercase when we validate it, example here:
[https://github.com/Bitcoin-ABC/bitcoin-abc/blob/master/src/cashaddr.cpp#L264](https://github.com/Bitcoin-ABC/bitcoin-abc/blob/master/src/cashaddr.cpp#L264)

so `ECASH` becomes `ecash` making it valid